### PR TITLE
[Yang] Add TUNNEL_TERM field to ACL_RULE table

### PR DESCRIPTION
**Pull request by vivekrnv**
#### Why I did it

Add support for user to match on TUNNEL_TERM flag in ACL_RULE table.

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111
- [ ] 202205
- [ ] 202211
- [ ] 202305

#### Tested branch (Please provide the tested image version)

- [ ] 
- [ ] 

#### A picture of a cute animal (not mandatory but encouraged)

### CODE REVIEW

**Review by oleksandrivantsiv — APPROVED** [review]

**Review by prsunny — DISMISSED** [review]

**Review by dgsudharsan — APPROVED** [review]

**Comment by vivekrnv** [comment]
@prsunny @dgsudharsan Can we merge this?

**Review by ganglyu — APPROVED** [review]

File: src/sonic-yang-models/yang-templates/sonic-acl.yang.j2 [hunk 1]
```diff
@@ -282,6 +282,11 @@ module sonic-acl {
 						pattern '0[xX][0-9a-fA-F]{1,2}/0[xX][0-9a-fA-F]{1,2}';
 					}
 				}
+
+				leaf TUNNEL_TERM {
+					description "Tunnel Termination Flag";
+					type boolean;
```
[the remark this answers is not in the document]
Comment (vivekrnv) [reply src/sonic-yang-models/yang-templates/sonic-acl.yang.j2:288]: Code only supports "true" or "false", so i think we can keep it boolean for now.

File: src/sonic-yang-models/doc/Configuration.md [hunk 2]
```diff
@@ -368,6 +369,7 @@ and migration plan
             "PRIORITY": "999",
             "PACKET_ACTION": "DROP",
             "SRC_IP": "1.1.1.1/32",
+            "TUNNEL_TERM": 'true"
```
[the remark this answers is not in the document]
Comment (vivekrnv) [reply src/sonic-yang-models/doc/Configuration.md]: Updated

**Review by wen587 — APPROVED** [review]

**Comment by oleksandrivantsiv** [comment]
@prsunny can we merge this PR?

**Review by qiluo-msft — APPROVED** [review]

**Pull request merged**